### PR TITLE
Abstracted the validators in FilterableListControls

### PR DESCRIPTION
**Pull request by jimmynotjim**
Abstracted the validators in FilterableListControls to make them testable

## Additions

- Adds new validators.js for reuse and testing.
- Adds unit tests for validators.js (passes 100% across the board).

## Changes

- Updates FilterableListControls to use new validators.
- Updates FilterableListControls to do all dom querying for fieldsets.
- Sneaks in linting error fixes.

## Testing

- `gulp lint:scripts && gulp test:unit:scripts`

## Review

- @sebworks 
- @anselmbradford 
- @KimberlyMunoz 

## Notes

- This is a spin-off of #1521 making testing the email regex change possible
- Updates in testing coverage to:

__Before__
```
=============================== Coverage summary ===============================
Statements   : 30.44% ( 589/1935 )
Branches     : 23.37% ( 154/659 )
Functions    : 23.19% ( 77/332 )
Lines        : 30.5% ( 586/1921 )
================================================================================
```
__After__
```
=============================== Coverage summary ===============================
Statements   : 31.21% ( 608/1948 )
Branches     : 27.06% ( 181/669 )
Functions    : 24.4% ( 81/332 )
Lines        : 31.33% ( 606/1934 )
================================================================================
```
## Todos

- Abstract out the other shared components of FilterableListControls and cf_validator
- More unit tests

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: cfgov/unprocessed/js/modules/util/validators.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ * @param {object} field         Field to test
+ * @param {object} currentStatus A previous tested status for the field
+ * @returns {object} empty if the field passes, msg and type if it failed
+ */
+function email( field, currentStatus ) {
+  var status = currentStatus || {};
+  var regex = [
+    '^[a-z0-9\u007F-\uffff!#$%&\'*+\/=?^_`{|}~-]+(?:\.[a-z0-9',
+    '\u007F-\uffff!#$%&\'*+\/=?^_`{|}~-]+)*@(?:[a-z0-9]',
+    '(?:[a-z0-9-]*[a-z0-9])?\.)+[a-z]{2,}$'
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/modules/util/validators.js]: As mentioned in the prior lint PR, could this just be a straight-up string? It's an array that's turned into a string with `join()` so the array bits be removed and the commas replaced with `+`?

File: test/unit_tests/modules/util/validators-spec.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
+'use strict';
+var chai = require( 'chai' );
+var expect = chai.expect;
+var jsdom = require( 'mocha-jsdom' );
+var validators = require(
+  '../../../../cfgov/unprocessed/js/modules/util/validators.js'
+);
+var ERROR_MESSAGES = require(
+  '../../../../cfgov/unprocessed/js/config/error-messages-config'
```
Comment (anselmbradford) [inline test/unit_tests/modules/util/validators-spec.js]: You could use a `var BASE_JS_PATH = '../../../../cfgov/unprocessed/js/';` constant at the top of the file, as is done in some other [specs](https://github.com/cfpb/cfgov-refresh/blob/flapjack/test/unit_tests/modules/util/assign-spec.js#L3).

File: cfgov/unprocessed/js/modules/util/validators.js (hunk 1 above)
Comment (jimmynotjim) [inline cfgov/unprocessed/js/modules/util/validators.js]: Multi-line object literal doesn't seem to work, updated to use a string.

File: test/unit_tests/modules/util/validators-spec.js (hunk 2 above)
Comment (jimmynotjim) [inline test/unit_tests/modules/util/validators-spec.js]: Ah yep, updated to match

**Comment by sebworks** [comment]
:+1:

**Comment by anselmbradford** [comment]
:+1:

**Pull request merged**